wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/WikibaseDataFetcher.java

```diff
@@ -305,6 +384,63 @@ public EntityDocument getEntityDocumentByTitle(String siteKey, String title)
 		return this.wbGetEntitiesAction.wbGetEntities(properties);
 	}
 
+	/**
+	 * Creates a map of titles to MediaInfoIds retrieved.
+	 *
+	 * @param numOfIds
+	 *            number of ids that should be retrieved
+	 * @param titles
+	 *            titles of the requested MediaInfoIds
+	 * @return map from titles for which data could be found to the MediaInfoIds
+	 *         that were retrieved
+	 * @throws IOException
+	 * @throws MediaWikiApiErrorException
+	 */
+	Map<String, MediaInfoIdValue> getMediaInfoIdMap(int numOfIds, String titles)
+			throws IOException, MediaWikiApiErrorException {
+		if (numOfIds == 0) {
+			return Collections.emptyMap();
+		}
+
+		Map<String, String> parameters = new HashMap<>();
+		parameters.put(ApiConnection.PARAM_ACTION, "query");
+		parameters.put("titles", titles);
+
+		Map<String, MediaInfoIdValue> result = new HashMap<>();
+
+		JsonNode root = connection.sendJsonRequest("POST", parameters);
+		if (!root.has("query")) return result; // empty query
+		JsonNode query = root.get("query");
+
+		// handle normalized situation
+		// for example, "Main_Page" will be normalized to "Main Page"
+		Map<String, String> normalizedMap = new HashMap<>();
+		if (query.has("normalized")) {
+			ArrayNode normalized = (ArrayNode) query.get("normalized");
+			Iterator<JsonNode> iterator = normalized.elements();
+			while (iterator.hasNext()) {
+				JsonNode next = iterator.next();
+				String from = next.get("from").asText();
+				String to = next.get("to").asText();
+				normalizedMap.put(to, from);
+			}
+		}
+
+		JsonNode pages = query.get("pages");
+		Iterator<Map.Entry<String, JsonNode>> iterator = pages.fields();
+		while (iterator.hasNext()) {
+			Map.Entry<String, JsonNode> page = iterator.next();
+			String pageId = page.getKey();
+			String normalizedTitle = page.getValue().get("title").textValue();
+			String title = normalizedMap.getOrDefault(normalizedTitle, normalizedTitle);
+			if (!"-1".equals(pageId)) { // "-1" means not found
```

Comment:
Could it be worth adding a test case for pages that are not found?

Reply:
Of course